core/src/main/java/com/novoda/noplayer/internal/exoplayer/AdvertPlaybackState.java

```diff
@@ -46,37 +42,9 @@ static AdvertPlaybackState from(List<AdvertBreak> advertBreaks, long advertBreak
         }
 
         adPlaybackState = adPlaybackState.withAdDurationsUs(advertBreaksWithAdvertDurations);
-        adPlaybackState = updateResumePositionInFirstGroup(adPlaybackState, advertBreakResumePositionMillis);
         return new AdvertPlaybackState(adPlaybackState, sortedAdvertBreaks);
     }
 
-    private static AdPlaybackState updateResumePositionInFirstGroup(AdPlaybackState state, long positionMillis) {
-        if (state.adGroupCount <= 0 || state.adGroups[0].count <= 0) {
-            return state;
-        }
-        long groupResumePosition = C.msToUs(positionMillis);
-        AdPlaybackState.AdGroup firstAdGroup = state.adGroups[0];
-
-        AdPlaybackState updatedState = state;
-        long playedAdvertDuration = 0;
-        for (int index = 0; index < firstAdGroup.count; index++) {
-            long durationWithCurrentAd = playedAdvertDuration + firstAdGroup.durationsUs[index];
-            if (durationWithCurrentAd <= groupResumePosition) {
-                updatedState = updatedState.withPlayedAd(0, index);
-                playedAdvertDuration += firstAdGroup.durationsUs[index];
-            }
-            if (groupResumePosition <= playedAdvertDuration) {
```

Comment:
This was the problem. The `playedAdvertDuration` is only updated when the resume position is greater than the adverts watched duration. However we don't break out if this first condition is met, so we can iterate through the adverts until we find one that matches our criteria. If we do this then the resume position updated on line 74 evaluates incorrectly.